### ISSUE TITLE
GetQuota() error system.argumentNullException

### ISSUE THREAD

**Issue by juansh20** [jstedfast/MailKit#433]
In version MailKit 1.10.0, When I call client.Inbox.GetQuota(); I get an error System.ArgumentNullException, In version MailKit 1.8.1 it works ok.

After, validate the client connection IMAP:

```csharp
if (client.IsConnected)
                {
                    if (client.Capabilities.HasFlag(ImapCapabilities.Quota))
                    {
                        if (client.Inbox.GetQuota() == null)
                        {
                            var quota = client.Inbox.GetQuota();
                            if (quota.StorageLimit.HasValue)
                            {
                                varResult.Add(quota.CurrentStorageSize.Value);
                                varResult.Add(quota.StorageLimit.Value);
                            }
                            else
                            {
                                varResult.Add(0);
                                varResult.Add(0);
                            }

                        }
                    }

```

**Comment by jstedfast** [comment]
what's the StackTrace on the exception say?

**Comment by jstedfast** [comment]
The problem is in your code:

```csharp
if (client.Inbox.GetQuota() == null)
{
    // if GetQuota() just returned null, why would you expect it to not return null the second time you call it?
    var quota = client.Inbox.GetQuota();
    if (quota.StorageLimit.HasValue) // <-- right here, quota is null and therefore the runtime throws a NullReferenceException
```

**Comment by juansh20** [comment]
Hello jstedfast:

I changed the code. it's the same error (value is null in client.Inbox.GetQuota()).

if (client.Capabilities.HasFlag(ImapCapabilities.Quota))
                    {
                        var quota = client.Inbox.GetQuota();
                        if (quota.StorageLimit.HasValue)
                        {
                            varResult.Add(quota.CurrentStorageSize.Value);
                            varResult.Add(quota.StorageLimit.Value);
                        }
                        else
                        {
                            varResult.Add(0);
                            varResult.Add(0);
                        }
                    }

StackTrace:

   en MailKit.FolderQuota..ctor(IMailFolder quotaRoot)
   en MailKit.Net.Imap.ImapFolder.UntaggedQuota(ImapEngine engine, ImapCommand ic, Int32 index)
   en MailKit.Net.Imap.ImapEngine.ProcessUntaggedResponse(CancellationToken cancellationToken)
   en MailKit.Net.Imap.ImapCommand.Step()
   en MailKit.Net.Imap.ImapEngine.Iterate()
   en MailKit.Net.Imap.ImapEngine.Wait(ImapCommand ic)
   en MailKit.Net.Imap.ImapFolder.GetQuota(CancellationToken cancellationToken)
   en EZSafe.Libraries.Shared.MailBox.CMailProcess.QuotaEmail(ArrayList AccountParam) 


In version MailKit 1.8.1 it works ok.

**Comment by jstedfast** [comment]
If you get a protocol log, is the quota root part of the response `NIL`?

**Comment by juansh20** [comment]
Connected to imap://XXX.XXXXX.XXXX:143/
S: * OK [CAPABILITY IMAP4rev1 LITERAL+ SASL-IR LOGIN-REFERRALS ID ENABLE IDLE NAMESPACE STARTTLS AUTH=PLAIN AUTH=LOGIN] Dovecot ready.
C: E00000000 AUTHENTICATE PLAIN AHBydWViYUBlenNoaXAucGUAUFJVRUJBODg=
S: E00000000 OK [CAPABILITY IMAP4rev1 LITERAL+ SASL-IR LOGIN-REFERRALS ID ENABLE IDLE SORT SORT=DISPLAY THREAD=REFERENCES THREAD=REFS THREAD=ORDEREDSUBJECT MULTIAPPEND URL-PARTIAL CATENATE UNSELECT CHILDREN NAMESPACE UIDPLUS LIST-EXTENDED I18NLEVEL=1 CONDSTORE QRESYNC ESEARCH ESORT SEARCHRES WITHIN CONTEXT=SEARCH LIST-STATUS BINARY MOVE NAMESPACE SPECIAL-USE COMPRESS=DEFLATE QUOTA] Logged in
C: E00000001 NAMESPACE
S: * NAMESPACE (("INBOX." ".")) NIL NIL
S: E00000001 OK Namespace completed.
C: E00000002 LIST (SPECIAL-USE) "" "*"
S: * LIST (\Junk) "." INBOX.Junk
S: * LIST (\Drafts) "." INBOX.Drafts
S: * LIST (\Trash) "." INBOX.Trash
S: * LIST (\Sent) "." INBOX.Sent
S: * LIST (\Archive) "." INBOX.Archive
S: E00000002 OK List completed (0.000 + 0.000 secs).
C: E00000003 GETQUOTAROOT INBOX
S: * QUOTAROOT INBOX storage=0
S: * QUOTA storage=0 (STORAGE 28257 256000)
S: E00000003 OK Getquotaroot completed (0.000 + 0.000 secs).

**Comment by jstedfast** [comment]
Thanks.

**Comment by jstedfast** [comment]
So the problem is that the quota root folder is `storage=0` which looks suspiciously like a non-folder.

Do you actually have a folder named `storage=0`?

**Comment by juansh20** [comment]
I don't have any folder named storage=0, maybe  some methods were changed in this version, because of version MailKit 1.8.1, when I debug this method, it's ok. 

protocol log in version 1.8.1

Connected to imap://xxx.xxxxxx.xxx:143/
S: * OK [CAPABILITY IMAP4rev1 LITERAL+ SASL-IR LOGIN-REFERRALS ID ENABLE IDLE NAMESPACE STARTTLS AUTH=PLAIN AUTH=LOGIN] Dovecot ready.
C: C00000000 AUTHENTICATE PLAIN AHBydWViYUBlenNoaXAucGUAUFJVRUJBODg=
S: C00000000 OK [CAPABILITY IMAP4rev1 LITERAL+ SASL-IR LOGIN-REFERRALS ID ENABLE IDLE SORT SORT=DISPLAY THREAD=REFERENCES THREAD=REFS THREAD=ORDEREDSUBJECT MULTIAPPEND URL-PARTIAL CATENATE UNSELECT CHILDREN NAMESPACE UIDPLUS LIST-EXTENDED I18NLEVEL=1 CONDSTORE QRESYNC ESEARCH ESORT SEARCHRES WITHIN CONTEXT=SEARCH LIST-STATUS BINARY MOVE NAMESPACE SPECIAL-USE COMPRESS=DEFLATE QUOTA] Logged in
C: C00000001 NAMESPACE
S: * NAMESPACE (("INBOX." ".")) NIL NIL
S: C00000001 OK Namespace completed.
C: C00000002 LIST (SPECIAL-USE) "" "*"
S: * LIST (\Junk) "." INBOX.Junk
S: * LIST (\Drafts) "." INBOX.Drafts
S: * LIST (\Trash) "." INBOX.Trash
S: * LIST (\Sent) "." INBOX.Sent
S: * LIST (\Archive) "." INBOX.Archive
S: C00000002 OK List completed (0.000 + 0.000 secs).
C: C00000003 GETQUOTAROOT INBOX
S: * QUOTAROOT INBOX storage=0
S: * QUOTA storage=0 (STORAGE 28257 256000)
S: C00000003 OK Getquotaroot completed (0.000 + 0.000 secs).
C: C00000004 LOGOUT
S: * BYE Logging out
S: C00000004 OK Logout completed.

**Comment by jstedfast** [comment]
The difference is that `FolderQuota`'s .ctor allows a `null` quotaRoot folder argument in 1.8, but in 1.10, it does not.

Since a lookup of the `storage=0` folder fails in both, the `quotaRoot` folder passed to the `FolderQuota` .ctor is `null`.

**Comment by jstedfast** [comment]
Technically 1.8 was also broken because you'd get back a `FolderQuota` with a `null` `QuotaRoot` property an thus would not be able to call `SetQuota()`

**Comment by jstedfast** [comment]
This is fixed now.

**Comment by juansh20** [comment]
Thanks jstedfast. Do you upgrade it in other version or version 1.10?. When can we download it?

**Comment by jstedfast** [comment]
I'll release a new nuget version this weekend. I'm not sure if I'll do 1.10.1 or 1.12 yet (depends on what else changes in the meantime).

**Comment by juansh20** [comment]
Thanks jstedfast.

It's a fantastic job that you are doing in Mailkit.

**Comment by jstedfast** [comment]
I released MailKit 1.10.1 with this fix to nuget.org
